spec/ataru/hakija/hakija_routes_spec.clj

```diff
@@ -417,7 +419,33 @@
     (it "should get application with hakuaika ended but hakukierros ongoing"
       (with-redefs [hakuaika/get-hakuaika-info hakuaika-ended-grace-period-passed-hakukierros-ongoing]
         (with-get-response "12345" resp
-          (should= 200 (:status resp))))))
+          (should= 200 (:status resp)))))
+
+    (it "should get application with hakuaika ended but field deadline extended"
+      (with-redefs [hakuaika/get-hakuaika-info         hakuaika-ended-grace-period-passed-hakukierros-ongoing
+                    field-deadline/get-field-deadlines (fn [_]
+                                                         [{:field-id      "b0839467-a6e8-4294-b5cc-830756bbda8a"
+                                                           :deadline      (.plusDays (DateTime/now) 1)
+                                                           :last-modified (DateTime/now)}])]
+        (with-get-response "12345" resp
+          (should= 200 (:status resp))
+          (should-not (->> (get-in resp [:body :form :content])
+                           util/flatten-form-fields
+                           (some #(when (= "b0839467-a6e8-4294-b5cc-830756bbda8a" (:id %)) %))
+                           :cannot-edit)))))
```

Comment:
Perinteikäs tuplanegaatio tässä assertissa :) Mutta ehkäpä `cannot-edit`:in refaktorointi `can-edit`:iksi on liian iso duuni, jotta maksaisi vaivaa.